compiler/src/org.graalvm.compiler.lir.aarch64/src/org/graalvm/compiler/lir/aarch64/AArch64AtomicMove.java

```diff
@@ -130,24 +135,40 @@ public AtomicReadAndAddOp(AArch64Kind kind, AllocatableValue result, Allocatable
             this.scratchValue2 = scratch2;
         }
 
+        private boolean isImmediateValue(Value value) {
+            if (isJavaConstant(value)) {
+                JavaConstant constant = asJavaConstant(value);
+                if (constant != null && constant.getJavaKind().isNumericInteger()) {
+                    int v = constant.asInt();
+                    if (v >= 0 && v < 4096) {
```

Comment:
Isn't there a method for this in the assembler?

Reply:
Yes, you are correct. I'll modify to use the existing one in AArch64MacroAssembler.isArithmeticImmediate().